Integrations/SignalSciences/SignalSciences.py

```diff
@@ -0,0 +1,1421 @@
+    human_readable['Source'] = ip_context['Source']
+    human_readable['Expiration date'] = ip_context['ExpiryDate'] if ip_context['ExpiryDate'] else "Not Set"
+    return human_readable
+
+
+def gen_context_for_add_to_whitelist_or_blacklist(response_data):
+    ip_context = {}
+    ip_context['ID'] = response_data.get('id', '')
+    ip_context['Note'] = response_data.get('note', '')
+    ip_context['Source'] = response_data.get('source', '')
+    ip_context['CreatedBy'] = response_data.get('createdBy', '')
+    ip_context['CreatedDate'] = response_data.get('created', '')
+    ip_context['ExpiryDate'] = response_data.get('expires', '')
+    return ip_context
+
+
+def add_ip_to_whitelist_or_blacklist(url, ip, note, expires=None):
+    check_ip_is_valid(ip)
+    data = {
+        'source': ip,
+        'note': note
+    }
+    if expires is not None:
+        data['expires'] = expires
+    res = http_request('PUT', url, data=data)
+    return res
+
+
+def get_all_sites_in_corp():
+    get_sites_request_response = get_sites()
+    data_of_sites_in_corp = get_sites_request_response.get('data', [])
+    return data_of_sites_in_corp
+
+
+def get_list_of_all_site_names_in_corp():
+    data_of_sites_in_corp = get_all_sites_in_corp()
+    list_of_all_sites_names_in_corp = []
+    for site_data in data_of_sites_in_corp:
+        site_name = site_data['name']
+        list_of_all_sites_names_in_corp.append(site_name)
+    return list_of_all_sites_names_in_corp
+
+
+def get_list_of_site_names_to_fetch():
+    list_of_site_names_to_fetch = None
+    if SITES_TO_FETCH:
+        list_of_site_names_to_fetch = SITES_TO_FETCH.split(',')
+    else:
+        list_of_site_names_to_fetch = get_list_of_all_site_names_in_corp()
+    return list_of_site_names_to_fetch
+
+
+def remove_milliseconds_from_iso(date_in_iso_format):
+    date_parts_arr = date_in_iso_format.split('.')
+    date_in_iso_without_milliseconds = date_parts_arr[0]
+    return date_in_iso_without_milliseconds
+
+
+def get_events_from_given_sites(list_of_site_names_to_fetch, desired_from_time_in_posix):
+    events_from_given_sites = []
+    for site_name in list_of_site_names_to_fetch:
+        fetch_from_site_response_json = get_events(siteName=site_name, from_time=desired_from_time_in_posix)
+
+        events_fetched_from_site = fetch_from_site_response_json.get('data', [])
+        events_from_given_sites.extend(events_fetched_from_site)
+    return events_from_given_sites
+
+
+def datetime_to_posix_without_milliseconds(datetime_object):
+    timestamp_in_unix_millisecond = date_to_timestamp(datetime_object, 'datetime.datetime')
+    posix_with_ms = timestamp_in_unix_millisecond
+    posix_without_ms = str(posix_with_ms).split(',')[0]
+    return posix_without_ms
+
+
+'''COMMANDS'''
+
+
+def test_module():
+    try:
+        url = SERVER_URL + 'corps'
+        http_request('GET', url)
+    except Exception as e:
+        raise Exception(e.message)
+
+    return_outputs(readable_output='ok')
```

Comment:
Does it work ok? I'm pretty sure we need to fill in the outputs

Reply:
will fix now